src/core/libmaven/PeakDetector.cpp

```diff
@@ -427,3 +430,58 @@ void PeakDetector::processSlices(vector<mzSlice *> &slices, string setName)
         }
     }
 }
+
+void PeakDetector::identifyFeatures(const vector<Compound*>& identificationSet)
+{
+    if (identificationSet.empty())
+        return;
+
+    GroupFiltering groupFiltering(mavenParameters);
+    vector<PeakGroup> toBeMerged;
+    auto iter = mavenParameters->allgroups.begin();
+    while(iter != mavenParameters->allgroups.end()) {
+        auto& group = *iter;
+        bool matchFound = false;
+        for (auto compound : identificationSet) {
+            float mz = 0.0f;
+            if (compound->formula.length()) {
+                int charge = mavenParameters->getCharge(compound);
+                mz = compound->adjustedMass(charge);
+            } else {
+                mz = compound->mass;
+            }
+            if (mzUtils::withinXMassCutoff(group.meanMz,
```

Comment:
Since targeted detection is done on the basis of compound->mz +/- cutoff, I would suggest sending `mz` as the first parameter and `group.meanMz` as the second parameter.

Reply:
Oh I had thought that `withinXMassCutoff` was checking proximity of two values independent of the order of the arguments. But, now that I see it, the check is whether the second mass lies in a small region around the first (in both directions). So, order does matter and I agree that query range should be created from compound's m/z. Making this change.